aiida/backends/manager.py

```diff
@@ -121,9 +121,13 @@ def load_backend_environment(self, profile, validate_schema=True):
     def _load_backend_environment(self):
         """Load the backend environment."""
 
-    @abc.abstractmethod
-    def reset_backend_environment(self):
+    @staticmethod
+    def reset_backend_environment():
```

Comment:
This shouldn't be removed from the django specific implementation. This method is for actually resetting the entire database environment, i.e. what is done by `_load_backend_environment`. The fact that we use SqlAlchemy for the query builder even for the Django environment is an implementation detail of the query builder not the whole database backend as a whole. So this is also not the correct place to reset the SqlA session for the query builder.

Reply:
The method never existed for `django`, but was empty. But you're saying that having the method actually _doing_ something for the `django` backend is wrong?

Reply:
And whether I reset the session for the QB or not shouldn't matter? If I want to reset the session, for any reason, I guess the place to do this should indeed be on the backend level, right?

Reply:
The method is intended to reset the database environment for that specific backend. This is necessary if you want to switch profile in the same interpreter. When you load a profile, as soon as the database backend is needed the `_load_backend_environment` is called. In the case of Django this will populate the `aiida.backends.djsite.settings` module with the correct database connection parameters and call `django.setup`. If you switch profile with another database, you have to "undo" i.e. reset this and do it again for the new profile settings. For SqlAlchemy this loading and resetting is simply the setting and resetting of the session.

Currently, the switching of profiles is not yet supported, which is why `reset_backend_environment` is not yet implemented for Django, because I haven't looked into it yet on how to do it. However, that doesn't mean it should be used for resetting the SqlAlchemy session when the querybuilder wants to. This is an implementation detail of the query builder and so should be performed there.

Reply:
Right! Thank you for the detailed explanation. I will find a work-around, specifically for the QB.